src/storage/txn/process.rs

```diff
@@ -261,41 +266,73 @@ pub(super) fn process_write_impl<S: Snapshot, L: LockManager>(
                 }
             }
             let mut txn = if scan_mode.is_some() {
-                MvccTxn::for_scan(snapshot, scan_mode, start_ts, !ctx.get_not_fill_cache())
+                MvccTxn::for_scan(
+                    snapshot,
+                    scan_mode,
+                    start_ts,
+                    !ctx.get_not_fill_cache(),
+                    pd_client,
+                )
             } else {
-                MvccTxn::new(snapshot, start_ts, !ctx.get_not_fill_cache())
+                MvccTxn::new(snapshot, start_ts, !ctx.get_not_fill_cache(), pd_client)
             };
 
             // Set extra op here for getting the write record when check write conflict in prewrite.
             txn.extra_op = extra_op;
 
+            let primary_key = Key::from_raw(&primary);
             let mut locks = vec![];
+            let mut async_commit_ts = TimeStamp::zero();
             for m in mutations {
+                let mut secondaries = &Some(vec![]);
+
+                if m.key() == &primary_key {
+                    secondaries = &secondary_keys
+                }
```

Comment:
If the mutation is a secondary lock, we shouldn't set `secondaries` to `Some` when async commit it not used.